guides/source/active_record_basics.md

```diff
@@ -374,3 +374,86 @@ and to roll it back, `rake db:rollback`.
 Note that the above code is database-agnostic: it will run in MySQL,
 PostgreSQL, Oracle and others. You can learn more about migrations in the
 [Active Record Migrations guide](migrations.html).
+
+Enum
+----
+
+Active Record `enum` allows you declare an attribute where the values map to
+integers in the database, but can be queried by name.
+
+You can define the enum on a model calling the `enum` macro.
+
+```ruby
+class Post < ActiveRecord::Base
+  enum status: [ :active, :archived ]
+end
+```
+
+This will add `enum` on the column `status`, where the active is represented as 0
+and the archived is represented as 1.
+
+It's also possible to explicitly map the relation between attribute and
+database integer with a `Hash`:
+
+```ruby
+class Post < ActiveRecord::Base
+  enum status: { active: 0, archived: 1 }
+end
+```
+
+You can set the default value from the database declaration, like:
+
+```ruby
+create_table :posts do |t|
+  t.column :status, :integer, default: 0
+end
+```
+
+Good practice is to let the first declared status be the default.
+
+You can change between the mapped statuses with bang methods:
+
+```ruby
+post.active!
+post.archived!
+```
+
+To check if a record is on a specific status use the question mark methods:
+
+```ruby
+post.active?   # => true
+post.archived? # => false
+```
+
+It is possible to assigns value directly to the `enum` column:
+
+```ruby
+post.status = "archived"
+```
+
+When reading the value it's returned the string and not the integer:
+
+```ruby
+post.status # => "archived"
+```
+
+Scopes based on the allowed values of the `enum` field will be provided as well.
+With the above example:
+
+```ruby
+Post.active
+Post.archived
+```
+
+Note that when an `Array` is used, the implicit mapping from the values to
+database integers is derived from the order the values appear in the array. In
+the example, `:active` is mapped to `0` as it's the first element, and
+`:archived` is mapped to `1`.
+Therefore, once a value is added to the enum array, its position in the array
+must be maintained, and new values should only be added to the end of the array.
+To remove unused values, the explicit `Hash` syntax should be used.
```

Comment:
This content is on the [API too](http://api.rubyonrails.org/classes/ActiveRecord/Enum.html).

I think the idea here is if we have something like:

```ruby
class Post < ActiveRecord::Base
  enum status: [ :active, :archived ]
end
```

And update the code to:

```ruby
class Post < ActiveRecord::Base
  enum status: [ :archived ]
end
```

Now all active posts(represented as 0) will became `archived` but before they are `active`, we can fix this with:

```ruby
class Post < ActiveRecord::Base
  enum status: { archived: 1 }
end
```

This way the `archived` will be always `1`.

I agree that we still can break things using the hash syntax, for example when using predicate methods on a removed map.

Do you prefer remove this?
> To remove unused values, the explicit `Hash` syntax should be used.